Writing-state-migrations.md

```diff
@@ -2,12 +2,11 @@ If your PR changes the properties of an exploration or state (or other structure
 
 ## Steps to follow while writing state migrations
 1. Make the necessary changes to the State class (or its descendant classes) to reflect the post-migration state structure.
-2. Make the necessary changes to the NEW_STATE_TEMPLATE in the constants.js file to reflect the post-migration state dict structure.
-3. Increment the CURRENT_STATE_SCHEMA_VERSION in the feconf.py file.
-4. Increment the CURRENT_EXP_SCHEMA_VERSION in the exp_domain.py file and similar changes in the question_domain.py file.
-5. Start with writing _convert_states_v(old_state_version)_dict_to_v(old_state_version + 1)_dict method in exp_domain.py files under Exploration class and in question_domain.py under Question class. In exp_domain.py, update the ```Exploration._migrate_to_latest_yaml_version``` method to use your conversion function (not required for question_domain.py, this is done automatically).
-6. Write a conversion function _convert_states_v(old_state_version)_dict_to_v(old_state_version + 1)_dict in draft_upgrade_services.py that makes appropriate upgrades to data that resides in ExplorationChange lists.
-7. Changing existing test files (Note: this list is not exhaustive. These files include tests that utilize a full state dictionary, but there are tests not listed here and tests that use a subset of state- like AnswerGroup, Interaction, etc.- that may have to be updated) :
+2. Increment the CURRENT_STATE_SCHEMA_VERSION in the feconf.py file.
+3. Increment the CURRENT_EXP_SCHEMA_VERSION in the exp_domain.py file and similar changes in the question_domain.py file.
+4. Start with writing _convert_states_v(old_state_version)_dict_to_v(old_state_version + 1)_dict method in exp_domain.py files under Exploration class and in question_domain.py under Question class. In exp_domain.py, update the ```Exploration._migrate_to_latest_yaml_version``` method to use your conversion function (not required for question_domain.py, this is done automatically).
+5. Write a conversion function _convert_states_v(old_state_version)_dict_to_v(old_state_version + 1)_dict in draft_upgrade_services.py that makes appropriate upgrades to data that resides in ExplorationChange lists. Here you have to check if the changes from your conversion function can be made to the draft_change_list available. If the changes can be made make sure to update the drafts otherwise raise `InvalidDraftConversionException` which will drop the changes in the drafts. If no changes needs to be done to the draft you can simply return the draft_change_list. You will be able to see examples for this in the previously mentioned functions.
```

Comment:
```suggestion
5. Write a conversion function `_convert_states_v{old_state_version}_dict_to_v{old_state_version + 1}_dict` in _draft_upgrade_services.py_ that makes appropriate upgrades to data that resides in `ExplorationChange` lists. Here you have to check if the changes that you do in your conversion function affect the `draft_change_list`, if yes, then make sure to update the drafts or raise `InvalidDraftConversionException` which will drop the changes in the drafts (this needs to be done in cases where changes to drafts would be too complicated). If no changes need to be done to drafts you can simply return the `draft_change_list`.
```

I've removed "You will be able to see examples for this in the previously mentioned functions." because I'm not sure what it means. Can you try to rephrase it?

Reply:
Updated, I think I do not have to add that line. Just wanted to let the user know that "they can check out previous conversion functions for reference" but they will be able to take reference from the example PRs.